api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java

```diff
@@ -214,28 +214,22 @@
 
   @Override
   public ResponseEntity<CohortChartDataListResponse> getCohortChartData(
-      String workspaceNamespace,
-      String workspaceId,
-      Long cohortReviewId,
-      String domain,
-      Integer limit) {
+      String workspaceNamespace, String workspaceId, Long cohortId, String domain, Integer limit) {
     int chartLimit = Optional.ofNullable(limit).orElse(DEFAULT_LIMIT);
     if (chartLimit < MIN_LIMIT || chartLimit > MAX_LIMIT) {
       throw new BadRequestException(
           String.format(
               "Bad Request: Please provide a chart limit between %d and %d.",
               MIN_LIMIT, MAX_LIMIT));
     }
-
     DbWorkspace dbWorkspace =
         workspaceAuthService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
             workspaceNamespace, workspaceId, WorkspaceAccessLevel.READER);
-    CohortReview cohortReview = cohortReviewService.findCohortReview(cohortReviewId);
-    DbCohort dbCohort =
-        cohortReviewService.findCohort(dbWorkspace.getWorkspaceId(), cohortReview.getCohortId());
+    DbCohort dbCohort = cohortReviewService.findCohort(dbWorkspace.getWorkspaceId(), cohortId);
+
     return ResponseEntity.ok(
         new CohortChartDataListResponse()
-            .count(cohortReview.getMatchedParticipantCount())
+            .count(cohortReviewService.participationCount(dbCohort))
```

Comment:
In practice the cohorts query can be very slow getting a count. For efficiency purposes I would try and read the review. If the review exists I would get the count -> cohortReview.getMatchedParticipantCount() otherwise if no review then call this method.

Reply:
I think i might have to change other stuff as well,  because i realized there could be multiple review set for a cohort, so in that case i should continue to pass cohortReview Id as well as an argument in this method and mark it as optional

Reply:
Currently we are only allowed to create 1 review per cohort.

Reply:
Turned i could create multiple review, because of any issue with local config . Have update the code.